sdk/src/main/java/com/uid2/sdk/network/RefreshResponse.kt

```diff
@@ -58,6 +58,11 @@ data class RefreshResponse(
             val body = json.optJSONObject("body")?.let { UID2Identity.fromJson(it) }
             val message = json.opt("message")?.toString()
 
+            // Check that if we've successfully refreshed, that we have a valid UID2Identity.
+            if (status == SUCCESS && body == null) {
```

Comment:
I found this unhandled case via the UTs. Seems sensible that we should validate we haven't parsed an invalid output (e.g. success but missing identity)

Reply:
Yep, this is a good thing to check for.  iOS does this implicitly via the Decrypt function... if there's any error found in the decryption process a specific UID2 error is thrown.

https://github.com/IABTechLab/uid2-ios-sdk/blob/835b73fb91543364b02394db565e9f40abf7ff3c/Sources/UID2/UID2Client.swift#L56-L58